app/src/main/java/com/example/hris/ui/fragments/leaves/FileLeaveFragment.kt

```diff
@@ -1,60 +1,164 @@
-            FileLeaveFragment().apply {
-                arguments = Bundle().apply {
-                    putString(ARG_PARAM1, param1)
-                    putString(ARG_PARAM2, param2)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        var leaveType = ""
+
+        val datePickerDialog = createDatePickerDialog()
+
+        binding.vacationLeave.setOnClickListener {
+            viewModel.type = "1"
+            leaveType = getString(R.string.vacation_leave)
+        }
+
+        binding.sickLeave.setOnClickListener {
+            viewModel.type = "2"
+            leaveType = getString(R.string.sick_leave)
+        }
+
+        binding.btnStartDate.setOnClickListener {
+            type = getString(R.string.start_date)
+            datePickerDialog.show()
+        }
+
+        binding.btnEndDate.setOnClickListener {
+            type = getString(R.string.end_date)
+            datePickerDialog.show()
+        }
+
+        binding.fileLeaveToolbar.btnDone.setOnClickListener {
+            viewModel.fileLeave(
+                binding.Spinner.selectedItemPosition + 1,
+                startDate,
+                endDate
+            )
+        }
+
+        binding.fileLeaveToolbar.btnCancel.setOnClickListener {
+            findNavController().navigate(FileLeaveFragmentDirections.actionFileLeaveFragmentToLeavesFragment())
+        }
+
+        viewModel.message.observe(viewLifecycleOwner) {
+            Toast.makeText(requireContext(), it, Toast.LENGTH_SHORT).show()
+        }
+
+        viewModel.loadingDialogState.observe(viewLifecycleOwner) {
+            mainViewModel.apiBool.value = it
+        }
+
+        viewModel.isSuccess.observe(viewLifecycleOwner) {
+            Toast.makeText(requireContext(), "Leave successfully filed", Toast.LENGTH_SHORT).show()
+            val action =
+                FileLeaveFragmentDirections.actionFileLeaveFragmentToFileLeaveSuccessFragment(
+                    Leaves(
+                        type = leaveType,
+                        dateFrom = startDate!!,
+                        dateTo = endDate,
+                        time = binding.Spinner.selectedItem.toString()
+                    )
+                )
+            findNavController().navigate(action)
+        }
+
+        binding.Spinner.onItemSelectedListener = object : OnItemSelectedListener {
+            override fun onItemSelected(
+                parent: AdapterView<*>?,
+                view: View?,
+                position: Int,
+                id: Long
+            ) {
+                if (position == 0) {
+                    setViewsVisibility(true)
+                } else {
+                    setViewsVisibility(false)
                 }
             }
+
+            override fun onNothingSelected(parent: AdapterView<*>?) {
+
+            }
+        }
+    }
+
+    private fun setViewsVisibility(boolean: Boolean) {
```

Comment:
Be more descriptive with field names. This could be renamed to `isVisible: Boolean`